### PR TITLE
Adding a Share button to campaign updates 

### PR DESCRIPTION
**Pull request by deadlybutter**
### What does this PR do?
📂  [Updating file structure for the components involved](https://github.com/DoSomething/phoenix-next/commit/b9bbdb992e408ece49245a7edfa1289dd0511c0b) 
🌐  [Change share component link](https://github.com/DoSomething/phoenix-next/commit/b0a8027b044db9b2c21f3b977abf90d80f2c6a14)
🆙  [Adding share button to campaign update](https://github.com/DoSomething/phoenix-next/commit/bb10c1caa7e504f648de3d5dbd5faefa6a0f0317)
💻  [Changing the Byline markup to include the share](https://github.com/DoSomething/phoenix-next/commit/dd2e1e297897a91793c0a2c40d1b6a0af50b7590)
🎨  [Styling the share button into a new variant per Luke suggestion](https://github.com/DoSomething/phoenix-next/commit/7a93b6d43c633ef8b9d69fd22ef9efd0dfa1b4d4)
🌎  [Building the share link for the update](https://github.com/DoSomething/phoenix-next/commit/772d4bd87f09d46ad40e16c23ac18cd488c1e7b2)
🔬 [Updated all relevant tests](https://github.com/DoSomething/phoenix-next/pull/332/commits/5f109da47e90be28c81ef02671fd182be52559b0) 

screen shot 2017-07-11 at 1 58 36 pm
screen shot 2017-07-11 at 1 58 29 pm

### Any background context you want to provide?
This PR does not address overriding the Facebook share card based on the campaign update, that is the next PR.

Also I need to update the tests, doing that now.

### CODE REVIEW

**Comment by deadlybutter** [comment]
apologizing the PR started off small but grew, I took the time to annotate all of the commits so it's a bit easier to follow the steps I took!

**Comment by weerd** [comment]
image 😱

File: resources/assets/components/Share/index.js [hunk 1]
```diff
@@ -1,40 +1,3 @@
...
-  parentSource: null,
-};
-
-Share.defaultMetadata = {
-  source: 'share button',
-  platform: 'facebook',
-};
-
-export default Share;
+export ShareContainer from './ShareContainer';
```
Comment (weerd) [inline resources/assets/components/Share/index.js]: Adds some of those sweet comment docblocks like you did in the `Byline`, etc.

Comment (deadlybutter) [inline resources/assets/components/Share/index.js]: Ah yes, will do 👍

**Comment by deadlybutter** [comment]
@weerd Totally open to new ideas for the share button stuff, different component maybe w/ a shared container?

**Pull request merged**